lib/helpers/buildURL.js

```diff
@@ -35,7 +35,10 @@ module.exports = function buildURL(url, params, paramsSerializer) {
     var parts = [];
 
     utils.forEach(params, function serialize(val, key) {
-      if (val === null || typeof val === 'undefined') {
+      if (typeof val === 'undefined') {
```

Comment:
val can't ever really be "not declared" so this should be `if (val === undefined)` i guess. not sure why it was ever `typeof`

Reply:
If `val` for some reason is `undefined` then `if (val === undefined)` would result in:
`Uncaught ReferenceError: val is not defined`
That's probably why it's a `typeof` check.

But if we know for certain that `val` cannot be `undefined` then we can just remove that portion of the if-statement.

Reply:
This is not true. `val` is declared as a parameter... it will always exist and can only have a value of `undefined`. There was never a reason to do a `typeof` here.

What you're talking about is if there literally wasn't a variable declaration of `val`, such as:

```ts
// start of file
console.log(val); // val is undefined and was never declared
```

but in this case, val is declared, it is a known parameter, it will always exist. it can have a value of undefined (if it isn't passed or if it is explicitly set to `undefined`).

Reply:
Sorry, my bad. I've updated it now. Should have tested it first... 😅